### PR TITLE
Added Bowtie2 module and test workflow

### PR DESCRIPTION
**Pull request by FelixKrueger**
Added is a functional module for `Bowtie2`.

The test requires the following command:
`./main.nf --genome E_coli`

The E. coli genome path is currently hard-coded, and needs to be adapted to a more final state. I have also generated a test dataset of 10,000 test reads (paired-end) with Sherman and the following command:

`Sherman -l 100 -n 10000 --genome /bi/scratch/Genomes/E_coli/ --paired -cr 0`. The data is available in the modules/test-data/tools/bowtie2/input (and output).

### CODE REVIEW

File: tools/bowtie2/main.nf [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		val (outdir)
+		val (bowtie2_args)
+		val (verbose)
+
+	output:
+	    path "*bam",  emit: bam
+		path "*stats.txt", emit: stats 
+
+	publishDir "$outdir/bowtie2",
+		mode: "copy", overwrite: true
```
Comment (ewels) [inline tools/bowtie2/main.nf]: tabs spaces

Comment (FelixKrueger) [reply tools/bowtie2/main.nf]: Ahh, thats's why it looks like that, in VS Code it all looks fine. I thought the whole indentation stuff is handled by the editors these days? Maybe not...

Comment (ewels) [reply tools/bowtie2/main.nf]: It should be a preference. I have plugins in atom to convert tabs to spaces and to highlight tabs, so it’s nice and clear and easy to stick to just spaces. It also highlights and auto-cleans trailing whitespace. The only downside is if you save a messy file then the diff shows that you edited half of the lines even if you only changed one line 🤭

File: tools/bowtie2/main.nf [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			readString = "-U " + reads
+		}
+		
+		index = params.genome["bowtie2"]
+		bowtie2_name = name + "_" + params.genome["name"]
+
+		println ("bowtie2 -x ${index} -p ${cores} ${bowtie2_options} ${readString}  2>${bowtie2_name}_bowtie2_stats.txt | samtools view -bS -F 4 -F 8 -F 256 -> ${bowtie2_name}_bowtie2.bam")
+		"""
+		module load bowtie2
+		module load samtools
```
Comment (ewels) [inline tools/bowtie2/main.nf:51]:
```suggestion
```

File: tools/bowtie2/main.nf [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+			readString = "-1 " + reads[0] + " -2 " + reads[1]
+		}
+		else {
+			readString = "-U " + reads
+		}
+		
+		index = params.genome["bowtie2"]
+		bowtie2_name = name + "_" + params.genome["name"]
+
+		println ("bowtie2 -x ${index} -p ${cores} ${bowtie2_options} ${readString}  2>${bowtie2_name}_bowtie2_stats.txt | samtools view -bS -F 4 -F 8 -F 256 -> ${bowtie2_name}_bowtie2.bam")
```
Comment (ewels) [inline tools/bowtie2/main.nf:48]:
```suggestion
```

**Pull request merged**